### PR TITLE
libvirtd: Add case to test restart libvirtd service

### PR DESCRIPTION
**Pull request by cliping**
RHEL7-103707 - [libvirtd] Restart libvirtd service bug 1450095

Signed-off-by: lcheng <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:
#avocado run --vt-type libvirt --vt-machine-type q35 libvirtd.libvirtd.restart_libvirtd --vt-connect-uri qemu:///system
JOB ID     : 69701812864eac7f4053fac238c1e112e0d12b13
JOB LOG    : /root/avocado/job-results/job-2022-03-17T11.14-6970181/job.log
 (1/1) type_specific.io-github-autotest-libvirt.libvirtd.libvirtd.restart_libvirtd: PASS (60.34 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 61.66 s


Test result in modular daemons mode:
#avocado run --vt-type libvirt --vt-machine-type q35 libvirtd.libvirtd.restart_libvirtd --vt-connect-uri qemu:///system
JOB ID     : 0f3840becae339d5e96df043f6e352c07966350e
JOB LOG    : /root/avocado/job-results/job-2022-03-17T11.03-0f3840b/job.log
 (1/1) type_specific.io-github-autotest-libvirt.libvirtd.libvirtd.restart_libvirtd: PASS (5.51 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 6.20 s

**Comment by cliping** [comment]
@yafu-1 , please help to review, thanks.

File: libvirt/tests/src/libvirtd/libvirtd.py [hunk 1]
```diff
@@ -58,15 +60,60 @@ def test_check_journal(libvirtd, test):
...
+
+    log_file = "/var/log/messages"
+    # clear messages
+    process.run("echo '' > %s" % log_file, shell=True)
+
+    libvirtd.restart()
+
+    error_msg = "gssapiv2_client_plug_init() failed in sasl_server_add_plugin()"
+    # Check libvirtd status
+    if libvirt_version.version_compare(5, 6, 0):
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirtd/libvirtd.py:87]: I assume we don't need determin libvirt version, but just judge whether utils_split_daemons.is_modular_daemon() return True or False

File: libvirt/tests/src/libvirtd/libvirtd.py [hunk 2]
```diff
@@ -58,15 +60,60 @@ def test_check_journal(libvirtd, test):
...
+    5) Check /var/log/messages
+
+    :param libvirtd: libvirtd object
+    :param test: test object
+    """
+
+    utils_libvirtd.Libvirtd("libvirtd-tls.socket").stop()
+    utils_libvirtd.Libvirtd("libvirtd-tcp.socket").stop()
+
+    log_file = "/var/log/messages"
```
Comment (yafu-1) [inline libvirt/tests/src/libvirtd/libvirtd.py:79]: tp-libvirt redirect daemon log to file and not exists in syslog anymore, so need to change the log file or log setting here.

**Review by yafu-1 — CHANGES_REQUESTED** [review]

File: libvirt/tests/src/libvirtd/libvirtd.py [hunk 3]
```diff
@@ -58,15 +60,60 @@ def test_check_journal(libvirtd, test):
...
+    if libvirt_version.version_compare(5, 6, 0):
+        if utils_split_daemons.is_modular_daemon():
+            cmd = "systemctl status virtqemud -l"
+        else:
+            cmd = "systemctl status libvirtd -l"
+    else:
+        cmd = "systemctl status libvirtd -l"
+    output = process.run(cmd, shell=True).stdout_text.strip()
+    if re.search(error_msg, output):
+        test.fail("Found error message in libvirtd status messages.")
```
Comment (yafu-1) [inline libvirt/tests/src/libvirtd/libvirtd.py:96]: It's better to output the error msg here.

File: libvirt/tests/src/libvirtd/libvirtd.py [hunk 4]
```diff
@@ -58,15 +60,60 @@ def test_check_journal(libvirtd, test):
...
+
+    # Check messages
+    if re.search(error_msg, log_file):
+        test.fail("Found error message in /var/log/messages.")
```
Comment (yafu-1) [inline libvirt/tests/src/libvirtd/libvirtd.py:100]: The same as above.

File: libvirt/tests/src/libvirtd/libvirtd.py (hunk 1 above)
Comment (cliping) [reply libvirt/tests/src/libvirtd/libvirtd.py]: Updated, thanks.

File: libvirt/tests/src/libvirtd/libvirtd.py (hunk 2 above)
Comment (cliping) [reply libvirt/tests/src/libvirtd/libvirtd.py]: Updated to daemon log, thanks.

File: libvirt/tests/src/libvirtd/libvirtd.py (hunk 3 above)
Comment (cliping) [reply libvirt/tests/src/libvirtd/libvirtd.py]: Updated.

**Review by chunfuwen — APPROVED** [review]
LGTM

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt libvirtd.libvirtd.check_journal --vt-machine-type q35 --vt-connect-uri qemu:///system
JOB ID     : ed65cba2f41c6517f3baaf605d67aebb04b16461
JOB LOG    : /var/lib/avocado/job-results/job-2022-05-09T00.29-ed65cba/job.log
 (1/1) type_specific.io-github-autotest-libvirt.libvirtd.libvirtd.check_journal: STARTED
 (1/1) type_specific.io-github-autotest-libvirt.libvirtd.libvirtd.check_journal: PASS (41.17 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2022-05-09T00.29-ed65cba/results.html
JOB TIME   : 42.76 s

**Review by yafu-1 — APPROVED** [review]
LGTM

**Pull request merged**